streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java

```diff
@@ -540,21 +540,18 @@ public void testStateGlobalThreadClose() throws Exception {
             waitForCondition(
                 () -> globalStreamThread.state() == GlobalStreamThread.State.DEAD,
                 "Thread never stopped.");
-            globalStreamThread.join();
```

Comment:
Why do we remove this line?

Reply:
Hi @mjsax, there will be no GlobalStreamThread thread triggered since we mock the globalStreamThread.start() to state change.

https://github.com/apache/kafka/blob/1854d4b8a11461b53b59fa109b95f2a4f5003997/streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java#L277-L283

Reply:
Ah. Thanks! Missed this part.